-DIUTEST_NO_ARGUMENT_DEPENDENT_LOOKUP failed test

```
[ RUN      ] StdArray.PrintTo
cxx_feature_tests.cpp:170: Succeeded.
{ 4-Byte object < 03 00 00 00 >, 4-Byte object < 01 00 00 00 >, 4-Byte object < 04 00 00 00 > }
logger_tests.hpp:72: error: Expected: strstr(printer_logger.c_str(), m_str.c_str()) != NULL
   Actual: strstr("cxx_feature_tests.cpp:170: Succeeded.
  { 4-Byte object < 03 00 00 00 >, 4-Byte object < 01 00 00 00 >, 4-Byte object < 04 00 00 00 > }
 ", 3, 1, 4) == NULL
[  FAILED  ] StdArray.PrintTo (0ms)
```

--- a/include/iutest_printers.hpp
+++ b/include/iutest_printers.hpp
@@ -249,6 +249,9 @@ inline void PrintTo(const T& value, iu_ostream* os) {
 }
 inline void PrintTo(bool b, iu_ostream* os)         { *os << (b ? "true" : "false"); }
 inline void PrintTo(const char* c, iu_ostream* os)  { *os << c; }
+#if defined(IUTEST_NO_ARGUMENT_DEPENDENT_LOOKUP)
+inline void PrintTo(int v, iu_ostream* os)  { *os << v; }
+#endif
 template<typename CharT, typename Traits, typename Alloc>
 inline void PrintTo(const ::std::basic_string<CharT, Traits, Alloc>& str, iu_ostream* os)   { *os << str.c_str(); }
 #if !defined(IUTEST_NO_FUNCTION_TEMPLATE_ORDERING)

--- a/test/cxx_feature_tests.cpp
+++ b/test/cxx_feature_tests.cpp
@@ -73,7 +73,7 @@ IUTEST(StringView, Compare)
 
 IUTEST(StringView, PrintTo)
 {
-    LogChecker ck("Hello");
+    PrintToLogChecker ck("Hello");
     ::std::string_view view = "Hello";
     IUTEST_SUCCEED() << ::iutest::PrintToString(view);
 }
@@ -103,12 +103,12 @@ IUTEST(Optional, Compare)
 IUTEST(Optional, PrintTo)
 {
     {
-        LogChecker ck("1234");
+        PrintToLogChecker ck("1234");
         ::std::optional<int> opt = 1234;
         IUTEST_SUCCEED() << ::iutest::PrintToString(opt);
     }
     {
-        LogChecker ck("nullopt");
+        PrintToLogChecker ck("nullopt");
         ::std::optional<int> opt = ::std::nullopt;
         IUTEST_SUCCEED() << ::iutest::PrintToString(opt);
     }
@@ -130,22 +130,22 @@ IUTEST(Variant, Compare)
 IUTEST(Variant, PrintTo)
 {
     {
-        LogChecker ck("1234");
+        PrintToLogChecker ck("1234");
         ::std::variant<int, float, ::std::string> v = 1234;
         IUTEST_SUCCEED() << ::iutest::PrintToString(v);
     }
     {
-        LogChecker ck("test");
+        PrintToLogChecker ck("test");
         ::std::variant<int, float, ::std::string> v("test");
         IUTEST_SUCCEED() << ::iutest::PrintToString(v);
     }
     {
-        LogChecker ck("monostate");
+        PrintToLogChecker ck("monostate");
         ::std::variant<std::monostate, int, float, std::string> v;
         IUTEST_SUCCEED() << ::iutest::PrintToString(v);
     }
     {
-        LogChecker ck("valueless_by_exception");
+        PrintToLogChecker ck("valueless_by_exception");
         ::std::variant<int, float, ::std::string> v = 0.2f;
         try
         {
@@ -165,7 +165,7 @@ IUTEST(Variant, PrintTo)
 
 IUTEST(StdArray, PrintTo)
 {
-    LogChecker ck("3, 1, 4");
+    PrintToLogChecker ck("3, 1, 4");
     ::std::array<int, 3> ar = { { 3, 1, 4 } };
     IUTEST_SUCCEED() << ::iutest::PrintToString(ar);
 }
@@ -186,7 +186,7 @@ IUTEST(StdArray, PrintTo)
 IUTEST(Any, PrintTo)
 {
     {
-        //LogChecker ck("1234");
+        //PrintToLogChecker ck("1234");
         ::std::any v = 1234;
         IUTEST_SUCCEED() << ::iutest::PrintToString(v);
     }
@@ -208,7 +208,7 @@ IUTEST(FileSystem, PathCompare)
 IUTEST(FileSystem, PathPrintTo)
 {
     {
-        LogChecker ck("/cxx_feature_tests.cpp");
+        PrintToLogChecker ck("/cxx_feature_tests.cpp");
         ::std::filesystem::path v = __FILE__;
         IUTEST_SUCCEED() << ::iutest::PrintToString(v);
     }
@@ -226,7 +226,7 @@ IUTEST(FileSystem, StatusCompare)
 IUTEST(FileSystem, StatusPrintTo)
 {
     {
-        LogChecker ck("regular: 0");
+        PrintToLogChecker ck("regular: 0");
         ::std::filesystem::file_status v = ::std::filesystem::status(__FILE__);
         IUTEST_SUCCEED() << ::iutest::PrintToString(v);
     }
@@ -235,19 +235,19 @@ IUTEST(FileSystem, StatusPrintTo)
 IUTEST(FileSystem, SpaceInfoPrintTo)
 {
     {
-        LogChecker ck("cpacity");
+        PrintToLogChecker ck("cpacity");
         ::std::filesystem::path path = __FILE__;
         ::std::filesystem::space_info v = ::std::filesystem::space(path.remove_filename());
         IUTEST_SUCCEED() << ::iutest::PrintToString(v);
     }
     {
-        LogChecker ck("free");
+        PrintToLogChecker ck("free");
         ::std::filesystem::path path = __FILE__;
         ::std::filesystem::space_info v = ::std::filesystem::space(path.remove_filename());
         IUTEST_SUCCEED() << ::iutest::PrintToString(v);
     }
     {
-        LogChecker ck("available");
+        PrintToLogChecker ck("available");
         ::std::filesystem::path path = __FILE__;
         ::std::filesystem::space_info v = ::std::filesystem::space(path.remove_filename());
         IUTEST_SUCCEED() << ::iutest::PrintToString(v);
@@ -270,7 +270,7 @@ IUTEST(FileSystem, DirectoryEntryPrintTo)
     {
         ::std::filesystem::path path = __FILE__;
         ::std::filesystem::path directory = path.remove_filename().append("testdata");
-        LogChecker ck(directory.generic_string());
+        PrintToLogChecker ck(directory.generic_string());
         ::std::filesystem::directory_entry x = *::std::filesystem::directory_iterator(directory);
         IUTEST_SUCCEED() << ::iutest::PrintToString(x);
     }
@@ -281,7 +281,7 @@ IUTEST(FileSystem, DirectoryIteratorPrintTo)
     {
         ::std::filesystem::path path = __FILE__;
         ::std::filesystem::path directory = path.remove_filename().append("testdata");
-        LogChecker ck(directory.generic_string());
+        PrintToLogChecker ck(directory.generic_string());
         ::std::filesystem::directory_iterator x = ::std::filesystem::directory_iterator(directory);
         IUTEST_SUCCEED() << ::iutest::PrintToString(x);
     }

--- a/test/list_test_tests.cpp
+++ b/test/list_test_tests.cpp
@@ -6,7 +6,7 @@
  *
  * @author      t.shirayanagi
  * @par         copyright
- * Copyright (C) 2013-2016, Takazumi Shirayanagi\n
+ * Copyright (C) 2013-2018, Takazumi Shirayanagi\n
  * This software is released under the new BSD License,
  * see LICENSE
 */
@@ -81,13 +81,8 @@ const char list_test_with_where_str[] =
 "  A\n"
 #endif
 #if IUTEST_HAS_PARAM_TEST
-#if !defined(IUTEST_NO_ARGUMENT_DEPENDENT_LOOKUP)
 "Test/ParamTest\n"
 "  A/0, where GetParam() = 42\n"
-#else
-"Test/ParamTest\n"
-"  A/0, where GetParam() = 4-Byte object < 2A 00 00 00 >\n"
-#endif
 #endif
 "";
 

--- a/test/logger_tests.hpp
+++ b/test/logger_tests.hpp
@@ -52,7 +52,6 @@ public:
 #endif
 
 #if !defined(IUTEST_USE_GTEST)
-
 class LogChecker
 {
     TestLogger printer_logger;
@@ -78,6 +77,35 @@ class LogChecker
 {
 public:
     explicit LogChecker(const char*) {}
+};
+#endif
+
+#if !defined(IUTEST_USE_GTEST) && !defined(IUTEST_NO_ARGUMENT_DEPENDENT_LOOKUP)
+class PrintToLogChecker
+{
+    TestLogger printer_logger;
+    ::std::string m_str;
+public:
+    explicit PrintToLogChecker(const char* str) : m_str(str)
+    {
+        ::iutest::detail::iuConsole::SetLogger(&printer_logger);
+    }
+    explicit PrintToLogChecker(const std::string& str) : m_str(str)
+    {
+        ::iutest::detail::iuConsole::SetLogger(&printer_logger);
+    }
+    ~PrintToLogChecker(void)
+    {
+        ::iutest::detail::iuConsole::SetLogger(NULL);
+        IUTEST_EXPECT_STRIN(m_str.c_str(), printer_logger.c_str());
+        printer_logger.clear();
+    }
+};
+#else
+class PrintToLogChecker
+{
+public:
+    explicit PrintToLogChecker(const char*) {}
 };
 #endif
 

--- a/test/matcher_tests.cpp
+++ b/test/matcher_tests.cpp
@@ -602,28 +602,16 @@ IUTEST(MatcherFailure, Each)
 
 IUTEST(MatcherFailure, ContainerEq)
 {
-#if !defined(IUTEST_NO_ARGUMENT_DEPENDENT_LOOKUP)
     CHECK_FAILURE( IUTEST_ASSERT_THAT(gb, ContainerEq(gc)), "ContainerEq: {1, 1, 1}" );
     CHECK_FAILURE( IUTEST_ASSERT_THAT(gb, ContainerEq(gc)), "Mismatch in a position 1: 1 vs 2" );
     CHECK_FAILURE( IUTEST_ASSERT_THAT(gb, ContainerEq(gc)), "Mismatch in a position 2: 1 vs 3" );
-#else
-    CHECK_FAILURE( IUTEST_ASSERT_THAT(gb, ContainerEq(gc)), "ContainerEq: {4-Byte" );
-    CHECK_FAILURE( IUTEST_ASSERT_THAT(gb, ContainerEq(gc))
-        , "Mismatch in a position 1: 4-Byte object < 01 00 00 00 > vs 4-Byte object < 02 00 00 00 >" );
-    CHECK_FAILURE( IUTEST_ASSERT_THAT(gb, ContainerEq(gc))
-        , "Mismatch in a position 2: 4-Byte object < 01 00 00 00 > vs 4-Byte object < 03 00 00 00 >" );
-#endif
     CHECK_FAILURE( IUTEST_ASSERT_THAT(ga, ContainerEq(gc)), "Mismatch element : 3 vs 10" );
     CHECK_FAILURE( IUTEST_ASSERT_THAT(gc, ContainerEq(ga)), "Mismatch element : 10 vs 3" );
 }
 
 IUTEST(MatcherFailure, Pointwise)
 {
-#if !defined(IUTEST_NO_ARGUMENT_DEPENDENT_LOOKUP)
     CHECK_FAILURE( IUTEST_ASSERT_THAT(gb, Pointwise(Eq(), gc)), "Pointwise: Eq: {1, 1, 1}" );
-#else
-    CHECK_FAILURE( IUTEST_ASSERT_THAT(gb, Pointwise(Eq(), gc)), "Pointwise: Eq: {4-Byte" );
-#endif
     CHECK_FAILURE( IUTEST_ASSERT_THAT(ga, Pointwise(Eq(), gc)), "Mismatch element : 3 vs 10" );
     CHECK_FAILURE( IUTEST_ASSERT_THAT(gc, Pointwise(Eq(), ga)), "Mismatch element : 10 vs 3" );
 }

--- a/test/printers_tests.cpp
+++ b/test/printers_tests.cpp
@@ -61,14 +61,22 @@ IUTEST(PrintToTest, Bar)
 IUTEST(PrintToTest, IutestAnyNotInitialized)
 {
     ::iutest::any a;
+#if !defined(IUTEST_NO_ARGUMENT_DEPENDENT_LOOKUP)
     LogChecker ck("empty");
+#else
+    LogChecker ck("8-Byte object < 00 00 00 00 00 00 00 00 >");
+#endif
     IUTEST_SUCCEED() << ::iutest::PrintToString(a);
 }
 
 IUTEST(PrintToTest, IutestAnyString)
 {
     ::iutest::any a = "any-test";
+#if !defined(IUTEST_NO_ARGUMENT_DEPENDENT_LOOKUP)
     LogChecker ck("any-test");
+#else
+    LogChecker ck("8-Byte object");
+#endif
     IUTEST_SUCCEED() << ::iutest::PrintToString(a);
 }
 
@@ -149,19 +157,11 @@ IUTEST(PrintToTest, Std)
     ::std::vector<int> v(a, a+(sizeof(a)/sizeof(a[0])));
 
     {
-#if !defined(IUTEST_NO_ARGUMENT_DEPENDENT_LOOKUP)
         LogChecker ck("0, 1");
-#else
-        LogChecker ck("4-Byte object < 00 00 00 00 >, 4-Byte object < 01 00 00 00 >");
-#endif
         IUTEST_SUCCEED() << ::iutest::PrintToString(p);
     }
     {
-#if !defined(IUTEST_NO_ARGUMENT_DEPENDENT_LOOKUP)
         LogChecker ck("{ 0, 1, 2 }");
-#else
-        LogChecker ck("{ 4-Byte object < 00 00 00 00 >, 4-Byte object < 01 00 00 00 >, 4-Byte object < 02 00 00 00 > }");
-#endif
         IUTEST_SUCCEED() << ::iutest::PrintToString(v);
     }
 }

